tools/automation/tests/__init__.py

```diff
@@ -42,6 +106,20 @@ def execute(args):
         output('Running in CI Mode')
         selected_modules = [('All modules', 'azure.cli', 'azure.cli'),
                             ('CLI Linter', 'automation.cli_linter', 'automation.cli_linter')]
+
+        modified_modules = _extract_top_level_modified_modules()
+        if any(base_mod in modified_modules for base_mod in ['core', 'testsdk', 'telemetry']):
+            pass    # if modified modules contains those 3 modules, run all tests
+        else:
+            test_paths = set()
+            for mod in modified_modules:
+                try:
+                    test_paths.add(os.path.normpath(test_index[mod]))
+                except KeyError:
+                    display("no tests found in module: {}".format(mod))
+            args.tests = test_paths
+
+            selected_modules = []
```

Comment:
In the beloew code, both selected_modules and args.tests are set, but here you cleaned selected_modules. What's the difference? 
https://github.com/Azure/azure-cli/blob/a590a3059cf10d4746f2540a4c5da1b87cc5cfda/tools/automation/tests/__init__.py#L133-L157

Reply:
The selected_modules is a list of tuple which consist of 3 elements, like:
https://github.com/Azure/azure-cli/blob/450f7511788f02f2a44eac252e58e4dfa1dfef6e/tools/automation/tests/__init__.py#L43-L44
The last element of the tuple is a Python Namesapce.

In the incremental test case, I identify module from modified files and then extract module name from it.
So the modified_modules is a list of Python module name like `core`, `vm`, `storage`.
Then I extract corresponding tests from testIndex.json which is generated from `get_test_index()`.

In the runner:
https://github.com/Azure/azure-cli/blob/450f7511788f02f2a44eac252e58e4dfa1dfef6e/tools/automation/tests/main.py#L15-L38

tests has higher priority than the module.
So, when I have the extract test need to run, no need to provide `selected_module`.